spark-plugins/src/main/java/co/cask/hydrator/plugin/spark/DecisionTreeRegressor.java

```diff
@@ -0,0 +1,243 @@
+  private DecisionTreeRegressorConfig config;
+  private Schema outputSchema;
+
+  @Override
+  public void configurePipeline(PipelineConfigurer pipelineConfigurer) throws IllegalArgumentException {
+    StageConfigurer stageConfigurer = pipelineConfigurer.getStageConfigurer();
+    Schema inputSchema = stageConfigurer.getInputSchema();
+    // if null, the input schema is unknown, or has multiple schemas.
+    if (inputSchema == null) {
+      outputSchema = null;
+      stageConfigurer.setOutputSchema(null);
+      return;
+    }
+    config.validate(inputSchema);
+    // otherwise, we have a constant input schema. Get the input schema and
+    // add a field to it, on which the prediction will be set
+    outputSchema = getOutputSchema(inputSchema);
+    stageConfigurer.setOutputSchema(outputSchema);
+  }
+
+
+  @Override
+  public JavaRDD<StructuredRecord> transform(SparkExecutionPluginContext context,
+                                             JavaRDD<StructuredRecord> input) throws Exception {
+    FileSet fileSet = context.getDataset(config.fileSetName);
+    Location modelLocation = fileSet.getBaseLocation().append(config.path);
+    if (!modelLocation.exists()) {
+      throw new IllegalArgumentException(String.format("Failed to find model to use for Regression. Location does " +
+                                                         "not exist: {}.", modelLocation));
+    }
+    final List<String> fields = config.getFeatureList(input.first().getSchema());
+
+    // load the model from a file in the model fileset
+    JavaSparkContext javaSparkContext = context.getSparkContext();
+    SparkContext sparkContext = JavaSparkContext.toSparkContext(javaSparkContext);
+    final DecisionTreeModel loadedModel = DecisionTreeModel.load(sparkContext, modelLocation.toURI().getPath());
+
+    JavaRDD<StructuredRecord> output = input.map(new Function<StructuredRecord, StructuredRecord>() {
+      @Override
+      public StructuredRecord call(StructuredRecord record) throws Exception {
+        List<Double> featureList = new ArrayList<>();
+        for (String field : fields) {
+          featureList.add(Double.valueOf(record.get(field).toString()));
+        }
+
+        Vector vector = Vectors.dense(Doubles.toArray(featureList));
+        double prediction = loadedModel.predict(vector);
+
+        return cloneRecord(record)
+          .set(config.predictionField, prediction)
+          .build();
+      }
+    });
+    return output;
+  }
+
+  /**
+   * Creates a builder based off the given record.
+   */
+  private StructuredRecord.Builder cloneRecord(StructuredRecord record) {
+    Schema schemaToUse = (outputSchema != null) ? outputSchema : getOutputSchema(record.getSchema());
+    List<Schema.Field> fields = new ArrayList<>();
+    fields.addAll(schemaToUse.getFields());
+    fields.addAll(Arrays.asList(Schema.Field.of(config.predictionField, Schema.of(Schema.Type.DOUBLE))));
```

Comment:
actually, schemaToUse isn't needed at all. You can just use outputSchema instead of building another schema for each record.